packages/medusa/src/loaders/module.ts

```diff
@@ -42,15 +47,50 @@ const registerModule = async (
     }
   }
 
+  if (
+    resolution.moduleDeclaration?.scope === MODULE_SCOPE.INTERNAL &&
+    resolution.moduleDeclaration?.resources === MODULE_RESOURCE_TYPE.SHARED
+  ) {
+    const moduleModels = loadedModule?.models || null
+    if (moduleModels) {
+      moduleModels.map((val: ClassConstructor<unknown> | EntitySchema) => {
+        const name = /function (\w+)/.exec(val.toString())?.[1]
+        if (!name) {
+          return
+        }
+
+        container.register({
+          [name]: asClass(val as ClassConstructor<unknown>),
+        })
+
```

Comment:
thought: could make sense to not allow resolution of the models directly. This would make it harder to abuse the models, but still make it available in `db_entities` for the core connection.

Reply:
That is a good point and will force the Module to be only accessible through the service layer